homeassistant/components/gogogate2/sensor.py

```diff
@@ -0,0 +1,103 @@
+"""Support for Gogogate2 temperature."""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+
+from homeassistant.const import (
+    CONF_USERNAME,
+    CONF_PASSWORD,
+    CONF_IP_ADDRESS,
+    CONF_NAME,
+    TEMP_CELSIUS,
+)
+from homeassistant.helpers.entity import Entity
+
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = "gogogate2"
+
+NOTIFICATION_ID = "gogogate2_temperature"
+NOTIFICATION_TITLE = "Gogogate2 Temperature"
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_IP_ADDRESS): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    }
+)
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the Gogogate2 component."""
+    from pygogogate2 import Gogogate2API as pygogogate2
+
+    temp_unit = hass.config.units.temperature_unit
+    
+    ip_address = config.get(CONF_IP_ADDRESS)
+    name = config.get(CONF_NAME)
+    password = config.get(CONF_PASSWORD)
+    username = config.get(CONF_USERNAME)
+
+    mygogogate2 = pygogogate2(username, password, ip_address)
```

Comment:
The integration needs to be refactored to create the api instance in the gogogate2 component, store the api in hass.data and load the platforms via discovery.

https://github.com/home-assistant/home-assistant/blob/7a171dae330fe37651eeef0d9000be44c1e585ee/homeassistant/helpers/discovery.py#L110